source/Handlebars/Compiler/Translation/Expression/PartialBinder.cs

```diff
@@ -22,6 +19,11 @@ private PartialBinder(CompilationContext context)
         {
         }
 
+        protected override Expression VisitBlockHelperExpression(BlockHelperExpression bhex)
+        {
+            return bhex;
+        }
+
```

Comment:
For some reason the PartialBinder shouldn't check within BlockHelpers or issues occur with `@partial-block`